From 12a48e39023698417dc31a215173d09458392f94 Mon Sep 17 00:00:00 2001
From: Venkatarama Avadhani <venkataramaa@google.com>
Date: Thu, 2 Jan 2025 14:18:07 +0530
Subject: [PATCH 26/37] Fix Integer Overflow in MPEG4Writer

When size is calculated in estimateMoovBoxSize, the max file size is
first multiplied by 6 and then divided by 1000. The multiplication
operation could result in an Integer Overflow. This change reorders the
multiplication and division to avoid the overflow.

Test: atest StagefrightRecorderTest
Bug: 383383170
Google: 3460372
Change-Id: I145620bda6930fd2b86a51857b3ea6dcfdf9480a
---
 media/libstagefright/MPEG4Writer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/libstagefright/MPEG4Writer.cpp b/media/libstagefright/MPEG4Writer.cpp
index 0bfb3ddb09..4e84786f06 100644
--- a/media/libstagefright/MPEG4Writer.cpp
+++ b/media/libstagefright/MPEG4Writer.cpp
@@ -883,7 +883,7 @@ int64_t MPEG4Writer::estimateMoovBoxSize(int32_t bitRate) {
 
     // Max file size limit is set
     if (mMaxFileSizeLimitBytes != 0 && mIsFileSizeLimitExplicitlyRequested) {
-        size = mMaxFileSizeLimitBytes * 6 / 1000;
+        size = mMaxFileSizeLimitBytes / 1000 * 6;
     }
 
     // Max file duration limit is set
-- 
2.34.1

